pywbem/_logging.py

```diff
@@ -0,0 +1,402 @@
+      logger = getLogger('pywbem.api')
+      logger.addHandler(handler)
+      logger.setLevel(logging.INFO)
+
+* This example uses the :func:`~py:logging.basicConfig` convenience function
+  that sets the same format and level as in the previous example, but for the
+  root logger. Therefore, it will output all log records, not just from this
+  package:
+
+  ::
+
+      import logging
+
+      format_string = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
+      logging.basicConfig(format=format_string, level=logging.DEBUG)
+"""
+from __future__ import absolute_import
+
+import logging
+import six
+
+# from ._constants import API_LOGGER_NAME
+API_LOGGER_NAME = 'pywbem.api'
+
+#: Name of logger for operantion call methods and their responses.
+#:
+LOG_OPS_CALLS_NAME = 'pywbem.ops'
+
+#: Name of logger for http information
+#:
+LOG_HTTP_NAME = 'pywbem.http'
+
+#: pywbem logger names. The following are the logger names that pywbem
+# recognizes.
+
+PYWBEM_LOG_COMPONENTS = ['ops', 'http', 'all']
+
+# possible log output destinations
+LOG_DESTINATIONS = ['file', 'stderr', 'none']
+
+LOG_DETAIL_LEVELS = ['all', 'min']
+
+DEFAULT_LOG_DETAIL_LEVEL = 'min'
+DEFAULT_LOG_LEVEL = 'debug'
+
+LOG_LEVELS = ['error', 'warning', 'info', 'debug']
+
+__all__ = ['PywbemLoggers', 'LOG_DESTINATIONS', 'LOG_LEVELS',
+           'PYWBEM_LOG_COMPONENTS', 'LOG_OPS_CALLS_NAME',
+           'LOG_DETAIL_LEVELS', 'DEFAULT_LOG_DETAIL_LEVEL', 'DEFAULT_LOG_LEVEL']
+
+
+class MetaPywbemLoggers(type):
+    """
+    This metaclass allows the definition of __str__ and __repr__ at the
+    class level in the subclass.
+    """
+    loggers = {}
+
+    def __str__(cls):
+        return 'PywbemLoggers(loggers={s.loggers!r})'.format(s=cls)
+
+    def __repr__(cls):
+        return 'PywbemLoggers({s.loggers!r})'.format(s=cls)
+
+
+# pylint: disable=no-init
+class PywbemLoggers(six.with_metaclass(MetaPywbemLoggers)):
+    """
+    Container for pywbem logger information when loggers are created. This
+    class is a singleton, there is only one set of data for a pywbem
+    instantiation.  It goal is to defined named loggers from data input but
+    it also records records about these named loggers in a dictionary
+
+    This is defined with only class level object and methods. This is an
+    easy way to create a singleton. However, at least in python 2.6, you
+    cannot make some magic methods work (ex.__getItem__)
+
+    There are two constructors for loggers:
+
+      PywbemLoggers.create_loggers - Creates one or more logger entries in
+      this class and also in the python logging class from an input string
+      with the format defined in the create_loggers method below.
+
+      PywbemLoggers.create_logger - Creates a single logger fromthe
+      separate pywbem logging attributes.
+
+    """
+    __metaclass__ = MetaPywbemLoggers
+
+    @classmethod
+    def reset(cls):
+        """Reset the logger dictionary. Used primarily in unittests"""
```

Comment:
This method has some built-in inconsistency, because it resets the dictionary but does not reset the logger objects.
Why does a unit test need to reset this dictionary? Is it because of the singleton design? There must be a better way to do that instead of providing a documented public method for that purpose.